src/transformers/trainer.py

```diff
@@ -213,7 +212,13 @@
 
 
 def _is_peft_model(model):
-    return is_peft_available() and isinstance(model, PeftModel)
+    classes_to_check = (PeftModel,) if is_peft_available() else ()
+    # Here we also check if the model is an instance of `PeftMixedModel` introduced in peft>=0.7.0: https://github.com/huggingface/transformers/pull/28321
+    if is_peft_available() and version.parse(importlib.metadata.version("peft")) >= version.parse("0.7.0"):
+        from peft import PeftMixedModel
+
+        classes_to_check = (*classes_to_check, PeftMixedModel)
+    return is_peft_available() and isinstance(model, classes_to_check)
```

Comment:
```suggestion
    if is_peft_available():
        classes_to_check = (PeftModel,) if is_peft_available() else ()
        # Here we also check if the model is an instance of `PeftMixedModel` introduced in peft>=0.7.0: https://github.com/huggingface/transformers/pull/28321
        if version.parse(importlib.metadata.version("peft")) >= version.parse("0.7.0"):
            from peft import PeftMixedModel
            classes_to_check = (*classes_to_check, PeftMixedModel)
         return isinstance(model, classes_to_check)
    return False
```
let's check for peft lib only once 😉